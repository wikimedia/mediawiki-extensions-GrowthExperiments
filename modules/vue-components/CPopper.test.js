const { mount } = require( '@vue/test-utils' );
const CPopper = require( './CPopper.vue' );

const renderComponent = ( { props, slots = { default: 'Some text' } } ) => {
	return mount( CPopper, {
		props,
		slots
	} );
};

/**
 * Mock the getBoundingClientRect and left
 * properties from the triggerRef.
 *
 * @param {number} clientHeight The height of the slotted trigger element
 * @param {number} left The x coordinate from the left side of the viewport
 * @param {{ left: number, right: number }} boundingRect A DOMRect-like object,
 * with left and right properties.
 * @return {{ clientHeight: number, left: number, getBoundingClientRect: Function }}
 */
const triggerRefMock = ( clientHeight, left, boundingRect ) => {

	return {
		clientHeight,
		left,
		getBoundingClientRect: jest.fn( () => boundingRect )
	};
};

describe( 'CPopper', () => {
	it( 'should render without a close button', () => {
		const wrapper = renderComponent( {
			props: {
				triggerRef: triggerRefMock( 300, 100, { left: 110, right: 290 } )
			}
		} );
		expect( wrapper.element ).toMatchSnapshot();
	} );

	it( 'should render above of the trigger element', () => {
		const wrapper = renderComponent( {
			props: {
				triggerRef: triggerRefMock( 300, 100, { left: 110, right: 290 } ),
				icon: 'Some icon',
				iconLabel: 'Some label',
				placement: 'above'
			}
		} );
		expect( wrapper.element ).toMatchSnapshot();
	} );
} );
