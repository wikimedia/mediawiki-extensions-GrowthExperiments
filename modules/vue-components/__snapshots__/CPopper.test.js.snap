// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CPopper should render with a close button 1`] = `
<div
  class="ext-growthExperiments-CPopper"
>
  <div
    class="ext-growthExperiments-CPopper__popover"
    style="top: 0px;"
    tabindex="0"
  >
    <div
      class="ext-growthExperiments-CPopper__popover__close-button-container"
    >
      <button
        aria-label="Some label"
        class="cdx-button cdx-button--action-default cdx-button--weight-quiet cdx-button--icon-only ext-growthExperiments-CPopper__popover__close-button"
      >
        
        <span
          class="cdx-icon cdx-icon--medium"
        >
          <svg
            aria-hidden="true"
            height="20"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <!---->
            <g>
              Some icon
            </g>
          </svg>
        </span>
        
      </button>
    </div>
    
    Some text
    
  </div>
</div>
`;

exports[`CPopper should render without a close button 1`] = `
<div
  class="ext-growthExperiments-CPopper"
>
  <div
    class="ext-growthExperiments-CPopper__popover"
    style="top: 0px;"
    tabindex="0"
  >
    <!--v-if-->
    
    Some text
    
  </div>
</div>
`;
